components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/internal/dao/impl/WebhookManagementDAOImpl.java

```diff
@@ -0,0 +1,320 @@
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            jdbcTemplate.withTransaction(template -> {
+                template.executeUpdate(WebhookSQLConstants.Query.DELETE_WEBHOOK,
+                        statement -> {
+                            statement.setString(WebhookSQLConstants.Column.UUID, webhookId);
+                            statement.setInt(WebhookSQLConstants.Column.TENANT_ID, tenantId);
+                        });
+                return null;
+            });
+        } catch (TransactionException e) {
+            throw new WebhookMgtServerException("Error while deleting the webhook in the system.", e);
+        }
+    }
+
+    @Override
+    public Webhook getWebhook(String webhookId, int tenantId) throws WebhookMgtException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            return jdbcTemplate.withTransaction(template -> {
+                Webhook webhook = template.fetchSingleRecord(WebhookSQLConstants.Query.GET_WEBHOOK_BY_ID,
+                        (resultSet, rowNumber) -> mapResultSetToWebhook(resultSet),
+                        statement -> {
+                            statement.setString(WebhookSQLConstants.Column.UUID, webhookId);
+                            statement.setInt(WebhookSQLConstants.Column.TENANT_ID, tenantId);
+                        });
+
+                if (webhook == null) {
+                    return null;
+                }
+
+                int internalWebhookId = getInternalWebhookIdByUuid(webhookId, tenantId);
+                List<String> events = getWebhookEvents(internalWebhookId);
+
+                // Build the Webhook without the secret
+                return new Webhook.BuilderWithoutSecret()
+                        .uuid(webhook.getUuid())
+                        .endpoint(webhook.getEndpoint())
+                        .description(webhook.getDescription())
+                        .tenantId(webhook.getTenantId())
+                        .eventSchemaName(webhook.getEventSchemaName())
+                        .eventSchemaUri(webhook.getEventSchemaUri())
+                        .status(webhook.getStatus())
+                        .createdAt(webhook.getCreatedAt())
+                        .updatedAt(webhook.getUpdatedAt())
+                        .eventsSubscribed(events)
+                        .build();
+            });
+        } catch (TransactionException e) {
+            throw new WebhookMgtServerException("Error while retrieving the webhook from the system.", e);
+        }
+    }
+
+    @Override
+    public List<String> getWebhookEvents(String webhookId, int tenantId) throws WebhookMgtException {
+
+        try {
+            int internalWebhookId = getInternalWebhookIdByUuid(webhookId, tenantId);
+            return getWebhookEvents(internalWebhookId);
```

Comment:
we can directly take from uuid and the tenant id right ? why we read the internal id and read back. we should have an index for uuid

Reply:
fixed with https://github.com/wso2/carbon-identity-framework/pull/6741/commits/153ddd354bdaf741877b9616f430bbae3d4b71b5